src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13979,5 +13948,196 @@ int32[] arg2
+                int MethodWithParams(params int[] xs) => xs.Length;
+                DelegateNoParams d7 = MethodWithParams;
+                DelegateWithParams d8 = MethodNoParams;
+                DelegateNoParams d9 = (params int[] xs) => xs.Length; // 3
+                DelegateWithParams d10 = (int[] xs) => xs.Length;
+
+                delegate int DelegateNoDefault(int x);
+                delegate int DelegateWithDefault(int x = 1);
+                delegate int DelegateNoParams(int[] xs);
+                delegate int DelegateWithParams(params int[] xs);
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (6,29): warning CS9501: Parameter 1 has default value '1' in lambda and '<missing>' in the target delegate type.
+                // DelegateNoDefault d4 = (int x = 1) => x; // 1
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "x").WithArguments("1", "1", "<missing>").WithLocation(6, 29),
+                // (7,31): warning CS9501: Parameter 1 has default value '2' in lambda and '1' in the target delegate type.
+                // DelegateWithDefault d5 = (int x = 2) => x; // 2
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "x").WithArguments("1", "2", "1").WithLocation(7, 31),
+                // (14,37): warning CS9502: Parameter 1 has params modifier in lambda and not in target delegate type.
+                // DelegateNoParams d9 = (params int[] xs) => xs.Length; // 3
+                Diagnostic(ErrorCode.WRN_ParamsArrayInLambdaOnly, "xs").WithArguments("1").WithLocation(14, 37));
+        }
+
+        [Fact]
+        public void DefaultsParamsConversion_Mix()
+        {
+            var source = """
+                D d1 = (int a, int b = 2, params int[] c) => { }; // 1, 2
+                void M(int a, int b = 2, params int[] c) { }
+                D d2 = M;
+                delegate void D(int x, int y, int[] z);
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (1,20): warning CS9501: Parameter 2 has default value '2' in lambda and '<missing>' in the target delegate type.
+                // D d1 = (int a, int b = 2, params int[] c) => { }; // 1, 2
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "b").WithArguments("2", "2", "<missing>").WithLocation(1, 20),
+                // (1,40): warning CS9502: Parameter 3 has params modifier in lambda and not in target delegate type.
+                // D d1 = (int a, int b = 2, params int[] c) => { }; // 1, 2
+                Diagnostic(ErrorCode.WRN_ParamsArrayInLambdaOnly, "c").WithArguments("3").WithLocation(1, 40),
+                // (1,41): error CS1737: Optional parameters must appear after all required parameters
+                // D d1 = (int a, int b = 2, params int[] c) => { }; // 1, 2
+                Diagnostic(ErrorCode.ERR_DefaultValueBeforeRequiredValue, ")").WithLocation(1, 41));
+        }
```

Comment:
Taking a step back, based on the change, I would expect the following scenarios to be covered:
1. method group conversion with params differences (error removed, covered by `ParamsArray_DelegateConversion`)
2. method group conversion with default value differences (error removed, covered by a few tests)
3. lambda conversion with params differences added a warning (has params or not, on both sides, so 4 variants, covered by `ParamsArray_LambdaConversion`). 
    Consider also including a case with bad `params` (for example `params int i` or `params int[] i, params int[] j`)
4. lambda conversion with default value differences added a warning (has no or same or different default or error value, on both sides, so 6 or 7 variants?). 
    Do we have all the cases are covered? It's hard to tell.
5. delegate creation expression with params or default value differences added a warning. 
    Do we have some tests for this (`new DelegateType(lambda)`)? I didn't see where.

Most of the conversion scenarios are in assignment. Consider adding a conversion scenario in invocation: `M(lambda)`, where the conversion succeeds but warns.

Reply:
> 3\. Consider also including a case with bad `params` (for example `params int i` or `params int[] i, params int[] j`)

These are covered in #64960.

> 4\. lambda conversion with default value differences added a warning (has no or same or different default or error value, on both sides, so 6 or 7 variants?).
>     Do we have all the cases are covered? It's hard to tell.

Some of these were already covered by Adam (see `MethodGroup_LambdaAssignment_*`), but I agree it's hard to tell whether all cases are covered. So, in this PR I added test `DefaultsParamsConversion_Spec` which has all these conversions in one place; now extended to also include more of the "match" cases: c1400efd5088c921c0fa7ae72e1f1ff11cd8a657

> 5\. delegate creation expression with params or default value differences added a warning.
>     Do we have some tests for this (`new DelegateType(lambda)`)? I didn't see where.

Good catch, there were no tests for this. Added: 08573b5af43412c6ba521726559ec5b5c788615f

> Most of the conversion scenarios are in assignment. Consider adding a conversion scenario in invocation: `M(lambda)`, where the conversion succeeds but warns.

Added: df1931572062d2f9c78151b634f9a28a2ce78731

Reply:
Thanks